mule-http-mock-extension/src/main/java/org/mule/test/extensions/http/mock/internal/server/HTTPMockServer.java

```diff
@@ -263,4 +265,109 @@ private byte[] toByteArray(InputStream input) throws IOException {
       return buffer.toByteArray();
     }
   }
+
+  /**
+   * Servlet dedicated to handling HTTP requests with the {@code Expect: 100-continue} header, with an added artificial delay.
+   *
+   * <p>
+   * This servlet simulates proper handling of the HTTP/1.1 "100 Continue" mechanism by first imposing a delay (e.g., 30 seconds)
+   * before fully reading the request body and processing the request. This is used for testing scenarios where a timeout or
+   * delayed response is needed.
+   * </p>
+   *
+   * <p>
+   * When a request is received at the mapping (e.g., "/expect-continue-delay"), the servlet:
+   * </p>
+   * <ol>
+   * <li>Checks for the {@code Expect: 100-continue} header and, if present, sends an interim 100 Continue status.</li>
+   * <li>Waits for a pre-configured delay period of 30,000 milli seconds.</li>
+   * <li>Reads the full request body and delegates processing to the custom {@link DelegateToFlowTransformer}.</li>
+   * <li>Generates an {@link HTTPMockServerResponse} with the final status, headers, and body.</li>
+   * <li>Tracks key processing timestamps which are exposed via custom response headers:
+   * <ul>
+   * <li>{@code X-Expect-Header-Time}: Time when the Expect header was received.</li>
+   * <li>{@code X-Continue-Sent-Time}: Time when the 100 Continue response was sent.</li>
+   * <li>{@code X-Body-Received-Time}: Time when the full request body was received.</li>
+   * </ul>
+   * </li>
+   * </ol>
+   *
+   * <p>
+   * Response status:
+   * </p>
+   * <ul>
+   * <li>100 Continue (if applicable)</li>
+   * <li>Final status from the {@code HTTPMockServerResponse}</li>
+   * <li>404 if no handler is registered for the requested path</li>
+   * </ul>
+   *
+   * <p>
+   * Response headers include:
+   * </p>
+   * <ul>
+   * <li>Standard response headers from the {@code HTTPMockServerResponse}</li>
+   * <li>Timestamp headers for diagnostics and verification</li>
+   * </ul>
+   *
+   * <p>
+   * This servlet is used for testing HTTP client behavior with the Expect/Continue mechanism under delayed response conditions in
+   * MUnit scenarios.
+   * </p>
+   */
+  private class DelayExpectContinueServlet extends HttpServlet {
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+      handle(req, resp);
+    }
+
+    private void handle(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+      long delayMillis = 30000;
```

Comment:
QQ: Why are we adding this delay? If a delay is needed, can this be shorter, so that we don't slow down our tests too much?